src/Datagrid/ProxyQueryInterface.php

```diff
@@ -21,81 +21,45 @@
 interface ProxyQueryInterface
 {
     /**
-     * @param string $name
-     * @param array  $args
+     * @param mixed[] $args
      *
      * @return mixed
      */
-    public function __call($name, $args);
+    public function __call(string $name, array $args);
 
     /**
-     * @param int|null $hydrationMode
+     * @param array<string, mixed> $params
      *
      * @return mixed
      */
-    public function execute(array $params = [], $hydrationMode = null);
+    public function execute(array $params = [], ?int $hydrationMode = null);
```

Comment:
I have to verify with a computer in DatagridBundle and storage bundle but I think these arguments are
- Possibly not used
- Too much doctrine related (even more after adding hydrationMode typehint)

We should use execute() signature or if we cant, at least use execute(array $options) signature (where options would have a parameter key and an hydratation key)

Reply:
I would need to revisit again, but I think they are used, atleast on doctrine orm one

Reply:
They are used here: https://github.com/sonata-project/SonataDoctrineORMAdminBundle/blob/3.x/src/Datagrid/ProxyQuery.php#L169

Reply:
It is indeed currently.

But you can see the signature of `PagerInterface:getResults()`.
```
/**
     * Returns an array of results on the given page.
     *
     * @return array
     */
    public function getResults();
```

And the implementation in DoctrineORMBundle
```
public function getResults($hydrationMode = Query::HYDRATE_OBJECT)
{
    return $this->getQuery()->execute([], $hydrationMode);
}
```

And there is multiple situation where we calling `execute` without argument.
For example, there is a method in the ModelManager called `executeQuery`, which does
```
$query->execute();
```

IMO, it seems possible to deprecate this arguments.
This ofc shouldn't be made in the same PR, but if you and @sonata-project/contributors agree, I'll create an issue.

Reply:
I do agree that this signature is weird and should be improved on a separate PR

Reply:
Issue is created, this is "solved"